src/core/useChannel.ts

```diff
@@ -16,31 +17,22 @@ import { usePusher } from "./usePusher";
  * ```
  */
 
-export const NO_CHANNEL_NAME_WARNING =
-  "No channel name passed to useChannel. No channel has been subscribed to.";
-
 export function useChannel<T extends Channel & PresenceChannel>(
   channelName: string | undefined
 ) {
   const { client } = usePusher();
-  const [channel, setChannel] = useState<T | undefined>();
-  useEffect(() => {
-    /** Return early if there's no client */
-    if (!client) return;
+  const [channel, setChannel] = useState<T>();
```

Comment:
I'm not sure if we need this useState, since subscribe and unsubscribe already changes the state of the Provider.

If someone confirms that is unnecessary, we can just use:
```ts
return getChannel(channelName)
```